mypartners/views.py

```
@@ -1941,244 +2012,250 @@ def api_convert_outreach_record(request):
-                                                   contact['notes'])
+        if create_partner:
+            partner_data['owner'] = user_company
+            partner_form = NuoPartnerForm(partner_data, company=user_company)
+        else:
+            api_errors.append(
+                "User does not have permission to create a partner.")
+
+    communication_record_form = None
+    communication_record = None
+    if 'communication_record' in data_object:
+        communication_record_data = data_object['communication_record']
+        communication_record_pk = communication_record_data.pop('pk', None)
+        create_communication_record = request.user.can(
+            user_company, u'create communication record')
+        if communication_record_pk:
+            communication_record = ContactRecord.get(
+                pk=communication_record_pk, partner__owner=user_company)
         else:
-            contact['contact'].notes = contact['notes']
-        contact['contact'].save()
-        outreach_record.contacts.add(contact['contact'])
-        if contact.get('location', None):
-            contact['location'].save()
-            contact['contact'].locations.add(contact['location'])
-        add_tags_to_object(contact['contact'], contact['tags'])
-
-        contact_record.partner = partner
-        contact_record.contact = contact['contact']
-        contact_record.save()
-        outreach_record.communication_records.add(contact_record)
-        add_tags_to_object(contact_record, contact_record_tags)
-
-    outreach_record.current_workflow_state = workflow_status
-    outreach_record.save()
+            if create_communication_record:
+                communication_record_form = NuoCommunicationRecordForm(
+                    communication_record_data, company=user_company)
+            else:
+                api_errors.append(
+                    "User does not have permission to create a " +
+                    "communication record.")
```

We should probably raise this error to an NUO access requirement at some point.. because without being able to create a comm record, there's no point in using NUO. Unlike Partner/Contact, which you could simply select existing.